### PR TITLE
Minor refactors

### PR DESCRIPTION
**Pull request by Jammjammjamm**
Some more minor refactoring.

### CODE REVIEW

File: lib/app/utils/web_driver.rb [hunk 1]
```diff
@@ -29,13 +29,11 @@ def run_script(json_script, start_url = nil)
...
         unless command['find_value'].nil?
           current_element = wait.until do
-            current = if !command['index'].nil?
-                        driver.find_elements(command['type'].to_sym => command['find_value'])
-                      else
-                        driver.find_element(command['type'].to_sym => command['find_value'])
-                      end
-
-            current if (current.is_a?(Array) && current.length >= (command['index'] || 0) && current[command['index']].displayed?) || (!current.is_a?(Array) && current.displayed?)
+            if !command['index'].nil?
```
Comment (radamson) [inline lib/app/utils/web_driver.rb]: Could replace this if..else..end with

```
if command['index'].nil?
  driver.find_element(command['type'].to_sym => command['find_value'])
else
  driver.find_elements(command['type'].to_sym => command['find_value'])
end
```

to get rid of the `!`

File: lib/app/models/sequence_result.rb [hunk 2]
```diff
@@ -34,6 +34,41 @@ class SequenceResult
...
+
+      def wait?
+        result == 'wait'
+      end
+
+      def failures
+        test_results.select(&:fail?)
+      end
+
+      def reset
```
Comment (arscan) [inline lib/app/models/sequence_result.rb]: Should this be `reset!`?

File: lib/app/models/testing_instance.rb [hunk 3]
```diff
@@ -74,22 +74,20 @@ def group_results(test_set_id)
         results = latest_results_by_case
 
         self.module.test_sets[test_set_id.to_sym].groups.each do |group|
-          result_details = group.test_cases.each_with_object(cancel: 0, pass: 0, skip: 0, fail: 0, error: 0, total: 0) do |val, hash|
-            next unless results.key?(val.id)
+          result_details = group.test_cases.each_with_object(Hash.new(0)) do |test_case, hash|
```
Comment (arscan) [inline lib/app/models/testing_instance.rb]: `Hash.new(0)`, good to know!

File: lib/app/models/testing_instance.rb [hunk 4]
```diff
@@ -99,38 +97,39 @@ def waiting_on_sequence
...
 
       def patient_id=(patient_id)
         return if patient_id.to_s == self.patient_id.to_s
 
-        resource_references.select { |ref| ref.resource_type == 'Patient' }
-        # Use destroy directly (instead of on each, so we don't have to reload)
-        resource_references.destroy
+        resource_references
+          .select { |ref| ref.resource_type == 'Patient' }
+          .destroy # Use destroy directly (instead of on each, so we don't have to reload)
```
Comment (arscan) [inline lib/app/models/testing_instance.rb]: Hmmm, was this a bug before, or did `select` mutate `resource_references`?  Seems like a bug.

File: lib/app/models/sequence_result.rb (hunk 2 above)
Comment (Jammjammjamm) [reply lib/app/models/sequence_result.rb]: Sure.

File: lib/app/models/testing_instance.rb (hunk 4 above)
Comment (Jammjammjamm) [reply lib/app/models/testing_instance.rb]: Oops, I didn't notice that I changed the behavior here. Do you know whether the intent is to delete all resources when the patient changes?

**Pull request merged**